code/XMLTester.xml

```diff
@@ -53,23 +53,23 @@
 <!--><-->
 
 <!-->CONF_QOS_STATIC_DISCOVERY_USERID<-->
-<publisher profile_name="publisher_xml_conf_static_discovery">
+<data_writer profile_name="publisher_xml_conf_static_discovery">
```

Comment:
Maybe also update the `profile_name` to something more according with `<data_writer>` tag.